TarSCM/scm/git.py

```diff
@@ -321,3 +321,9 @@ def prepare_working_copy(self):
     # no cleanup is necessary for git
     def cleanup(self):
         pass
+
+    def check_url(self):
+        """check if url is a remote url"""
+        if not re.match("^(https?|ftps?|git)://", self.url):
```

Comment:
@M0ses , that check seems to be missing ssh. Might not be common, but I guess is often used in private OBS deployments.

Reply:
I did not add it for purpose.
In OBS, ssh should not be used. 
Without special preparation (known_hosts), it will break or services get stuck.
Running within a disabled service via osc might make sense in some cases.
I could add it if there is a special use case for it, but ATM I see no reason for this.

Another note: If I would add it for ssh, also syntax like 
```git@github.com:openSUSE/obs-service-tar_scm```
might be wanted.

As I already tried to point out, I would not recommend ssh as transport protocol for services at all.

Reply:
I see your point.

But this particular change is breaking existing private OBS deployments using ssh transport. (Some of my colleagues run into this already when the picked up the latest tar-scm service package build from the openSUSE:Tools repo).

So I would argue to not break the user experience on this, to keep it somewhat backward compatible. Not breaking existing workflows/deployments is more important then making assumption on special preparations which people might miss (e.g. know_hosts).

At least in the past I got in touch with various such private OBS deployments using Git SSH transport. Especially OBS deployments which date back when there was no smart-http Git backend available. Private OBS deployment might have not migrated all the services files for maintenance-releases, from SSH to a HTTPS transport (or might still not offer smart-http Git backends).

I created this PR: https://github.com/openSUSE/obs-service-tar_scm/pull/253 ... which is not covering the ...
   git@github.com:openSUSE/obs-service-tar_scm

... case.


CC' @cschneemann